src/Atmos/Model/prog_prim_conversion.jl

```diff
@@ -74,11 +76,12 @@ function prognostic_to_primitive!(
     prim::Vars,
     prog::Vars,
     e_int::AbstractFloat,
```

Comment:
Can/should we remove this from the arg list?

Reply:
Happy you mentioned this! I have now moved this low-level functions to be computed within the functions for each type, in terms of the aux vars. This is a tad more verbose, but avoids possible inputs that are incoherent with respect to the auxiliary state, since this function is accessible outside here. As an example, we were actually using the functions `(bl, bl.moisture, ...) ` in the test. I have made it now so that the test calls the high-level dispatcher, as we should do everywhere in the model.

Reply:
I worry a bit about using the `aux` state here (#1648), but we should be okay if we only use `new_thermo_state`